apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -341,28 +374,82 @@ impl ApplyTo for PathSelection {
     }
 }
 
+impl ApplyTo for JSLiteral {
+    fn apply_to_path(
+        &self,
+        data: &JSON,
+        vars: &VarsWithPathsMap,
+        input_path: &InputPath<JSON>,
+        errors: &mut IndexSet<ApplyToError>,
+    ) -> Option<JSON> {
+        match self {
+            Self::String(s) => Some(JSON::String(s.clone().into())),
+            Self::Number(n) => {
+                if let Ok(JSON::Number(n)) = serde_json_bytes::serde_json::from_str(n) {
+                    Some(JSON::Number(n))
+                } else {
+                    errors.insert(ApplyToError::new(
+                        format!("Invalid number {}", n).as_str(),
+                        input_path.to_vec(),
+                    ));
```

Comment:
Can we guarantee at parse time that literal numbers are valid numbers instead? Since we're going to have to catch this with validation in composition anyway—definitely don't want to fail at runtime on something we _can_ catch statically.

Reply:
This code went away with https://github.com/apollographql/router/pull/5762/commits/ab8bcd76b7ffb88354ea17dac92ffb8e1631855b. 🎉